File: majority-element/v2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        last_num = None
        last_count = None
        for num in nums:
            if num != last_num:
                last_num = num
                last_count = 1
            else:
                last_count += 1

            if last_count > len(nums) / 2:
                return last_num

        return -1


# Runtime: 196 ms, faster than 61.77% of Python3 online submissions for Majority Element.
# Memory Usage: 15.5 MB, less than 81.81% of Python3 online submissions for Majority Element.

File: longest-substring-without-repeating-characters/v2.go
package v2

func lengthOfLongestSubstring(s string) int {
	lastOccurrence := map[rune]int{}
	substringStartIdx := 0
	longestSubstring := 0

	var lastOccurrenceIdx, thisLen int
	var ok bool
	for i, c := range s {
		lastOccurrenceIdx, ok = lastOccurrence[c]
		if ok && lastOccurrenceIdx >= substringStartIdx {
			substringStartIdx = lastOccurrenceIdx + 1
		}
		lastOccurrence[c] = i

		thisLen = i - substringStartIdx + 1
		if thisLen > longestSubstring {
			longestSubstring = thisLen
		}
	}

	return longestSubstring
}

// Runtime: 10 ms, faster than 62.17% of Go online submissions for Longest Substring Without Repeating Characters.
// Memory Usage: 3.2 MB, less than 45.79% of Go online submissions for Longest Substring Without Repeating Characters.
